extends CharacterBody2D


const SPEED = 200.0
const JUMP_VELOCITY = -500.0


func _physics_process(delta: float) -> void:
	# Add the gravity.
	if not is_on_floor():
		velocity += get_gravity() * delta

	# Handle jump.
	if Input.is_action_just_pressed("jump") and is_on_floor():
		velocity.y = JUMP_VELOCITY

	var direction := Input.get_axis("move_left", "move_right")
	if direction:
		velocity.x = direction * SPEED
		$AnimatedSprite2D.play("run")
		if direction == -1:
			$AnimatedSprite2D.flip_h = true
		else:
			$AnimatedSprite2D.flip_h = false
	else:
		velocity.x = move_toward(velocity.x, 0, SPEED)
		$AnimatedSprite2D.play("idle")
		
	#if not is_on_floor():
		#$AnimatedSprite2D.play("jump")
		

	move_and_slide()

#@onready var animated_sprite_2d = $AnimatedSprite2D
#
## Called when the node enters the scene tree for the first time.
#func _ready() -> void:
	#animated_sprite_2d.play("run"); # Replace with function body.


# Called every frame. 'delta' is the elapsed time since the previous frame.
func _process(delta: float) -> void:
	pass


func _on_portal_body_entered(body: Node2D) -> void:
	get_tree().change_scene_to_file("res://levels/level_1/level_icon_1.tscn") # Replace with function body.
